src/components/ec-button-group/ec-button-group.spec.js

```diff
@@ -0,0 +1,218 @@
+          items: [
+            { text: 'Yes', value: 'yes' },
+            { text: 'Maybe' },
+          ],
+        });
+        expect(errorSpy).toHaveBeenCalled();
+        expect(errorSpy.mock.calls[0][0]).toContain('Invalid prop: custom validator check failed for prop "items"');
+      });
+    });
+
+    it(':items - should throw an error when one or more items doesn\'t have value and text property', () => {
+      withMockedConsole((errorSpy) => {
+        mountButtonGroup({
+          items: [
+            { value: 'yes' },
+            { text: 'Maybe' },
+          ],
+        });
+        expect(errorSpy).toHaveBeenCalled();
+        expect(errorSpy.mock.calls[0][0]).toContain('Invalid prop: custom validator check failed for prop "items"');
+      });
+    });
+  });
+
+  describe(':modal', () => {
+    it('should render the button group and change the value when you click on the current not selected', () => {
+      const wrapper = mountButtonGroupAsTemplate(
+        '<ec-button-group v-model="value" :items="items"></ec-button-group>',
+        {},
+        {
+          data() {
+            return {
+              value: 'yes',
+              items: [
+                { text: 'Yes', value: 'yes' },
+                { text: 'No', value: 'no' },
+              ],
+            };
+          },
+        },
+      );
+      expect(wrapper.vm.value).toBe('yes');
+      expect(wrapper.find('.ec-btn--outline').text()).toBe('No');
+      wrapper.find('.ec-btn--outline').trigger('click');
+      expect(wrapper.find('.ec-btn--outline').text()).toBe('Yes');
+      expect(wrapper.vm.value).toBe('no');
+    });
+
+    it('should render the button group without any selected and change the value when you click on the current not selected', () => {
+      const wrapper = mountButtonGroupAsTemplate(
+        '<ec-button-group v-model="value" :items="items"></ec-button-group>',
+        {},
+        {
+          data() {
+            return {
+              value: '',
+              items: [
+                { text: 'Yes', value: 'yes' },
+                { text: 'No', value: 'no' },
+              ],
+            };
+          },
+        },
+      );
+      expect(wrapper.vm.value).toBe('');
+      wrapper.find('.ec-btn--outline').trigger('click');
+      expect(wrapper.find('.ec-btn--outline').text()).toBe('No');
+      expect(wrapper.vm.value).toBe('yes');
+    });
+
+    it('should render the button group and not change when click on the disabled item', () => {
+      const wrapper = mountButtonGroupAsTemplate(
+        '<ec-button-group v-model="value" :items="items"></ec-button-group>',
+        {},
+        {
+          data() {
+            return {
+              value: 'yes',
+              items: [
+                { text: 'Yes', value: 'yes' },
+                { text: 'No', value: 'no', disabled: true },
+              ],
+            };
+          },
+        },
+      );
+      expect(wrapper.vm.value).toBe('yes');
```

Comment:
I'd keep this test scenario, but target the disabled button and checking the value of the model doesn't change after clicking the disabled button